metashare/api/sf_org_changes.py

```diff
@@ -51,10 +53,10 @@ def run_retrieve_task(user, scratch_org, project_path, desired_changes):
         default_package_directory
```

Comment:
It seems like the above lines setting `package_directories` and `default_package_directory` can be removed? They're also slightly different than the new code in `jobs._get_valid_target_directories` -- maybe this should be moved over there?

Reply:
There's a slight difference in the logic between this code and `get_valid_target_directories`; I'd like to ask @davisagli about it before changing.

David:

`get_valid_target_directories` operates on the assumptions described in https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_ws_config.htm and presumes the first value in `"packageDirectories"` is the default if there is no explicit default. The logic above in _this_ function seeks an explicitly marked default, and depends on that. Should we be tracking the _explicit_ default as opposed to the implicit default? Does that distinction still matter here? Thanks!

Reply:
@wlonk 

> `get_valid_target_directories` operates on the assumptions described in https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_ws_config.htm and presumes the first value in `"packageDirectories"` is the default if there is no explicit default.

I don't see anything in the docs that say to use the first value:

> If you have only one path, we assume it’s the default, so you don’t have to explicitly set the default parameter. If you have multiple paths, you must indicate which one is the default.

So I assume we should check for an explicit default, or if there's only one value then treat it as the default.

Reply:
The docs say "if you have more than one value, you _must_ have one and only one marked as 'default'," so that means that if you have none marked as default, you _must_ have only one, and then we can take the first-and-only as the default. This means that if we have anything the "source" key of the dict that `get_valid_target_directories` returns, the first one will always be the default. I am not sure how this interacts with the logic already here and would like a sanity check on it, because I don't know what assumptions the logic here is built on.

Reply:
Understood. I think given the docs we can reasonably assume that the intention is to get the default directory, either explicitly marked or the only (or first, defensively) one.